crates/sui-graphql-rpc/src/types/protocol_config.rs

```diff
@@ -19,69 +17,36 @@ pub(crate) struct ProtocolConfigFeatureFlag {
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub(crate) struct ProtocolConfigs {
-    pub configs: Vec<ProtocolConfigAttr>,
-    pub feature_flags: Vec<ProtocolConfigFeatureFlag>,
     pub protocol_version: u64,
+    pub feature_flags: Vec<ProtocolConfigFeatureFlag>,
+    pub configs: Vec<ProtocolConfigAttr>,
 }
 
-#[allow(unreachable_code)]
-#[allow(unused_variables)]
 #[Object]
 impl ProtocolConfigs {
-    async fn configs(&self, ctx: &Context<'_>) -> Result<Option<Vec<ProtocolConfigAttr>>> {
-        Ok(Some(
-            // TODO: implement DB counterpart without using Sui SDK client
-            ctx.data_provider()
-                .fetch_protocol_config(None)
-                .await?
-                .configs,
-        ))
+    async fn protocol_version(&self) -> Result<u64> {
+        // TODO: implement DB counterpart without using Sui SDK client
+        Ok(self.protocol_version)
     }
 
-    async fn feature_flags(
-        &self,
-        ctx: &Context<'_>,
-    ) -> Result<Option<Vec<ProtocolConfigFeatureFlag>>> {
-        Ok(Some(
-            // TODO: implement DB counterpart without using Sui SDK client
-            ctx.data_provider()
-                .fetch_protocol_config(None)
-                .await?
-                .feature_flags,
-        ))
+    async fn feature_flags(&self) -> Result<Option<Vec<ProtocolConfigFeatureFlag>>> {
```

Comment:
any complaints if i modify these types to be Vec<ProtocolConfigFeatureFlag> instead of Option<Vec<Option<ProtocolConfigFeatureFlag>>>> as currently defined in the draft_target_schema